pkg/kb_agent/cronjobs/job.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright (C) 2022-2024 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package cronjobs
+
+import (
+	"context"
+	"time"
+
+	"github.com/apecloud/kubeblocks/pkg/constant"
+	"github.com/apecloud/kubeblocks/pkg/kb_agent/handlers"
+	"github.com/apecloud/kubeblocks/pkg/kb_agent/util"
+)
+
+type Job interface {
+	Start()
+	Stop()
+}
+
+type CommonJob struct {
+	Name             string
+	Ticker           *time.Ticker
+	TimeoutSeconds   int
+	PeriodSeconds    int
+	SuccessThreshold int
+	FailureThreshold int
+	FailedCount      int
+	ReportFrequency  int
+	Do               func() error
+}
+
+func NewJob(name string, cronJob *util.CronJob) Job {
+	job := &CommonJob{
+		Name:             name,
+		TimeoutSeconds:   60,
+		PeriodSeconds:    60,
+		SuccessThreshold: 1,
+		FailureThreshold: 3,
+		ReportFrequency:  60,
+	}
+
+	if cronJob.PeriodSeconds != 0 {
+		job.PeriodSeconds = cronJob.PeriodSeconds
+	}
+
+	if cronJob.SuccessThreshold != 0 {
+		job.SuccessThreshold = cronJob.SuccessThreshold
+	}
+
+	if cronJob.FailureThreshold != 0 {
+		job.FailureThreshold = cronJob.FailureThreshold
+	}
+
+	if name == constant.RoleProbeAction {
+		return NewCheckRoleJob(*job)
+	}
+
+	job.Do = job.do
```

Comment:
looks strange, change to this ?
```
job.Do = func () error {
	_, err := handlers.Do(context.Background(), job.Name, nil)
	return err
}
```

Reply:
CommonJob.Start is intended to be a shared function between common jobs and special jobs, while Job.Do is used for specific implementations for individual jobs. Since Go supports composition but not inheritance, Do is defined as a field of Job to allow different implementations to be assigned.